### ISSUE TITLE
Documentation renders incorrectly

### ISSUE THREAD

**Issue by 3ee6a0c8-3049-4ab4-b820-144dc843796e** [python/cpython#75918]
BPO | [31737](https://bugs.python.org/issue31737)
--- | :---
Nosy | @JulienPalard, @gibson042, @tirkarthi

<sup>*Note: these values reflect the state of the issue at the time it was migrated and might not reflect the current state.*</sup>

<details><summary>Show more details</summary><p>

GitHub fields:
```python
assignee = None
closed_at = <Date 2018-09-22.21:21:07.690>
created_at = <Date 2017-10-09.17:49:43.204>
labels = ['3.7', 'docs']
title = 'Documentation renders incorrectly'
updated_at = <Date 2018-09-22.21:21:07.689>
user = 'https://github.com/gibson042'
```

bugs.python.org fields:
```python
activity = <Date 2018-09-22.21:21:07.689>
actor = 'mdk'
assignee = 'docs@python'
closed = True
closed_date = <Date 2018-09-22.21:21:07.690>
closer = 'mdk'
components = ['Documentation']
creation = <Date 2017-10-09.17:49:43.204>
creator = 'gibson042'
dependencies = []
files = []
hgrepos = []
issue_num = 31737
keywords = []
message_count = 5.0
messages = ['303988', '303990', '303996', '326109', '326116']
nosy_count = 4.0
nosy_names = ['docs@python', 'mdk', 'gibson042', 'xtreak']
pr_nums = []
priority = 'normal'
resolution = 'fixed'
stage = 'resolved'
status = 'closed'
superseder = None
type = None
url = 'https://bugs.python.org/issue31737'
versions = ['Python 2.7', 'Python 3.4', 'Python 3.5', 'Python 3.6', 'Python 3.7']
```

</p></details>

**Comment by 3ee6a0c8-3049-4ab4-b820-144dc843796e** [comment]
The content at docs.python.org seems to be inserting language-dependent "smart quotes" in code blocks, which mangles backslashes and sequences like `'''`. Observed at https://docs.python.org/3/reference/lexical_analysis.html#string-and-bytes-literals , which renders

  longstring      ::=  “”’” longstringitem* “”’” | ‘”“”’ longstringitem* ‘”“”’

instead of

  longstring      ::=  "'''" longstringitem* "'''" | '"""' longstringitem* '"""'

and

  stringescapeseq ::=  “" \<any source character\>

instead of

  stringescapeseq ::=  "\\" \<any source character\>

, and looks even worse in other languages:

  longstring      ::=   » »” » longstringitem*  » »” » | “ »«  »” longstringitem* “ »«  »”

  longstring      ::=  「」』」 longstringitem* 「」』」 | 『」「」』 longstringitem* 『」「」』


Running `make html` locally produces the desired output, so whatever's going on appears specific to the public site.

**Comment by JulienPalard** [comment]
We spotted some quoting probleme in the french translation too, and I still didn't had time to look at it, typically here: https://docs.python.org/fr/3/library/itertools.html#itertool-functions

where we're getting:

  10.1.1. Fonctions d”itertool¶

instead of:

  10.1.1. Fonctions d'itertool¶

**Comment by JulienPalard** [comment]
Tested locally with different versions of sphinx-build:

- 1.3.3 does not produce the issue
- 1.6.2 (used on docs.python.org) produces the issue
- 1.6.3 does not produce the issue
- 1.6.4 does not produce the issue

Reading the changelog between 1.6.2 and 1.6.3 lead to https://github.com/sphinx-doc/sphinx/issues/3824 itself leading to
https://github.com/sphinx-doc/sphinx/commit/f99fe20e507210bd7db8665b8f234f1fe25c8100

So it looks fixed in 1.6.3 and 1.6.4, I'm running a full local build (using docsbuild-scripts) using sphinx 1.6.4 to see if we can upgrade.
